packages/block-library/src/utils/get-transformed-metadata.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockType } from '@wordpress/blocks';
+
+/**
+ * Transform the metadata attribute with only the values and bindings specified by each transform.
+ * Returns `undefined` if the input metadata is falsy.
+ *
+ * @param {Object} metadata      Original metadata attribute from the block that is being transformed.
+ * @param {Array}  fromBlockName Name of the original block that is being transformed.
+ * @param {Object} toBlockName   Name of the final block after the transformation.
+ * @return {Object|undefined} New metadata object only with the relevant properties.
+ */
+export function getTransformedMetadata( metadata, fromBlockName, toBlockName ) {
```

Comment:
It's probably enough to pass an optional second param that is an object with mapping between bindings:

```js
getTransformedMetadata( metadata, { content: 'text' } );
```

or with a callback:

```js
getTransformedMetadata( metadata, ( { bindings: { content } } ) => ( { bindings: { text: content } } );
```

Reply:
This makes sense 🙂 

I added the callback in [this commit](https://github.com/WordPress/gutenberg/pull/59179/commits/5e3eda0efe3b2b2cedd2aafe078ef0a6427e6779). I preserved the `toBlockName` property because it is needed to get the block supports.

Reply:
All good 👍🏻